packages/shared/src/node-apis/redis.ts

```diff
@@ -89,19 +88,17 @@ export default class RedisClient {
   async getDeposits(chain: Chain, asset: Asset, address: string) {
     const parsedAddress = chain === 'Polkadot' ? ss58ToHex(address) : address;
     const key = `deposit:${chain}:${parsedAddress}`;
-    const value = await this.client.get(key);
-    return value
-      ? deposits
-          .parse(JSON.parse(value))
-          .filter((deposit) => deposit.asset.asset === asset)
-          .sort(sortDepositAscending)
-      : [];
+    const deposits = await this.client.lrange(key, 0, -1);
+    return deposits
+      .map((deposit) => depositSchema.parse(deposit))
+      .filter((deposit) => deposit.asset === asset)
+      .sort(sortDepositAscending);
```

Comment:
as this is a linked list and we are using `rpush` to store it, i think we would not need to sort

Reply:
yeah we probably never needed to but we're not guaranteed to always be using `rpush` to store it